### PR TITLE
Valhalla - Scaffolding

### PR DESCRIPTION
**Pull request by chitalian**
### IT WORKS
```bash
$ curl https://mrm39x8vja.us-west-2.awsapprunner.com/healthcheck-db
{"status":"healthy :)","dataBase":"healthy :)"}
```

# Introducing - Valhalla
Valhalla contains all of the resources for helicone's new backend for storing and retrieving data. 

This PR contains the scaffolding for all the services within Terraform and updates the local development setup to work with the services. 

# The changes
image

With this change we are now able to easily support the following
- Multi tenant (with just deploying custom Valhalla services within customer clouds and then pointing our services to it)
- Horizontal scalability with sharding and read replicas
- Dedicated servers for egress and ingress to support larger payloads than Vercel edge functions

## Key Changes
- Docker compose for local setup + run_local.sh script to simply all the services to a single command
- Terraform script for setting up Valhalla
- `Jawn` and `Slay` are using yarn workspaces + Lerna to share code between the two services while also serving as a mono-repo. This will be a test to see how easy it is sharing code between services and might be worth doing in the future. 

# What is next
- Add DrizzleORM to Jawn/Slay
- Implement Jawn and Slay to behave as expected
- Design a backfill migration
- Add CI to automatically build and deploy new containers for Jawn and Slay
- Add CI to automatically run Drizzle migrations (this one is going to be hard lol - might just want to do it manually for now)

## Small tasks
- Clean up main.tf to have better descriptions and improve security policies (see if we can isolate the 10.11.x.x subnet to have /24 rather than /16 discoverability

### CODE REVIEW

File: valhalla/packages/jawn/src/db/client.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { Client } from "pg";
+
+// const auroraCreds = process.env.AURORA_CREDS || "";
```
Comment (colegottdank) [inline valhalla/packages/jawn/src/db/client.ts:3]: I assume this in coming later?

File: valhalla/packages/jawn/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import express from "express";
+import morgan from "morgan";
+import { Client } from "pg";
+import { hello } from "helicone-shared-ts";
+
+const app = express();
+const environment = process.env.ENV || "development";
+
+// for logs
+app.use(morgan("combined"));
```
Comment (colegottdank) [inline valhalla/packages/jawn/src/index.ts:11]: Are we going to hookup morgan logs with another service like datadog etc?

File: valhalla/packages/slay/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
+// // src/index.ts
```
Comment (colegottdank) [inline valhalla/packages/slay/src/index.ts:1]: Same question for this, coming soon?

File: valhalla/terraform/main.tf [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+
+
+  database_name = local.database_name
+
+  name            = local.name
+  engine          = "aurora-postgresql"
+  engine_version  = "14.7"
+  master_username = "root"
+  storage_type    = "aurora-iopt1"
+  instances = {
```
Comment (colegottdank) [inline valhalla/terraform/main.tf:39]: 3 instances?

File: valhalla/packages/jawn/src/db/client.ts (hunk 1 above)
Comment (chitalian) [reply valhalla/packages/jawn/src/db/client.ts:3]: yep yep

File: valhalla/packages/jawn/src/index.ts (hunk 2 above)
Comment (chitalian) [reply valhalla/packages/jawn/src/index.ts:11]: yeah or cloudwatch

File: valhalla/packages/slay/src/index.ts (hunk 3 above)
Comment (chitalian) [reply valhalla/packages/slay/src/index.ts:1]: yessir

File: valhalla/terraform/main.tf (hunk 4 above)
Comment (chitalian) [reply valhalla/terraform/main.tf:39]: 🤷 I just copied it from the example

**Pull request merged**